aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -796,17 +794,16 @@ Tensor view_dtype(const Tensor& self, ScalarType dtype) {
       self.storage_offset());
```

Comment:
This check is using `storage_offset` and `size` rather than the `sym` equivalents.